src/components/Messages.js

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Image, ScrollView, Text, View } from "react-native";
+
+import styles from "./styles";
+
+const Messages = ({ messages, user }) => {
+  const renderMessages = () => {
+    let currentUser = "";
+
+    return messages?.map(
+      ({ message, user: { _id: userId, name, photoURL }, _id }) => {
+        let isSameUser = currentUser === userId;
+        if (!isSameUser) {
+          currentUser = userId;
+        }
+
+        const isMyComment = user.uid === userId;
+        const showUserInfo = !isSameUser && !isMyComment;
+
+        return (
+          <View style={styles.rowContainer({ isMyComment })}>
+            {showUserInfo && (
+              <Image source={{ uri: photoURL }} style={styles.userPhoto} />
+            )}
+            <View
+              key={_id}
+              style={styles.messageContainer({
+                addMargin: !showUserInfo && !isMyComment,
+                isMyComment,
+                isSameUser,
+              })}
+            >
+              {showUserInfo && <Text style={styles.userName}>{name}</Text>}
+              <Text style={styles.message}>{message}</Text>
+            </View>
+          </View>
+        );
+      }
+    );
+  };
+
+  return (
+    <ScrollView
```

Comment:
I think right now the best option to handle a list of messages from a chat app it's w/ FlashList since it's designed to improve the performance and optimize the way it renders multiple elements